adapters/src/transport/file.rs

```diff
@@ -0,0 +1,331 @@
+use super::{InputConsumer, InputEndpoint, InputTransport};
+use crate::PipelineState;
+use anyhow::{Error as AnyError, Result as AnyResult};
+use crossbeam::sync::{Parker, Unparker};
+use num_traits::FromPrimitive;
+use serde::Deserialize;
+use serde_yaml::Value as YamlValue;
+use std::{
+    borrow::Cow,
+    fs::File,
+    io::{BufRead, BufReader},
+    sync::{
+        atomic::{AtomicU32, Ordering},
+        Arc,
+    },
+    thread::{sleep, spawn},
+    time::Duration,
+};
+
+const SLEEP_MS: u64 = 200;
+
+/// `InputTransport` implementation that reads data from file.
```

Comment:
is the file assumed to not change after it is opened?

Reply:
It can be appended to, but truncation or deletion will cause the endpoint to fail.  This seems like a reasonable behavior.